staging/src/k8s.io/apiserver/pkg/server/healthz.go

```diff
@@ -99,7 +104,9 @@ func (s *GenericAPIServer) AddDelayedHealthzChecks(delay time.Duration, checks .
 		s.healthzChecks = append(s.healthzChecks, delayedHealthCheck(check, s.healthzClock, s.maxStartupSequenceDuration))
 	}
 
-	return s.AddReadyzChecks(checks...)
+	s.readyzLock.Lock()
+	defer s.readyzLock.Unlock()
+	return s.addReadyzChecks(checks...)
```

Comment:
this change is not needed, is it?

Reply:
the lock should be acquired prior to calling AddReadyzChecks().
Hence the change.